src/mpi/topo/dims_create.c

```diff
@@ -65,6 +65,14 @@ typedef struct Factors {
 PMPI_LOCAL int MPIR_Dims_create_init(void);
 PMPI_LOCAL int MPIR_Dims_create_impl(int nnodes, int ndims, int dims[]);
 
+/* Define MPICH_MPI_FROM_PMPI if weak symbols are not supported to build
```

Comment:
Sorry probably just a lack of knowledge from my side but I don't understand how the commit message relates to the changes in this patch.

Reply:
Without weak symbols, we build 2 libraries - libmpi and libpmpi. The preprocessor macro `MPICH_MPI_FROM_PMPI` is used to compile files with `MPI_*` functions twice - once from libmpi and once for libpmpi. The libpmpi version has the actual guts and functionality, since the user is not allowed to override it. Here I move some of those guts (i.e. globals) to just the libpmpi version of this file.

Reply:
Ok sorry now I get it, thanks for the explanation. In that case this looks good

Reply:
👍 can you signoff and merge?